source/common/event/dispatcher_impl.cc

```diff
@@ -306,5 +325,22 @@ void DispatcherImpl::touchWatchdog() {
   }
 }
 
+void DispatcherImpl::pushTrackedObject(const ScopeTrackedObject* object) {
```

Comment:
Please add the following 2 ASSERTs as sanity checks:

ASSERT(isThreadSafe());
ASSERT(object !=nullptr);